giskard/llm/client/mistral.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, Sequence
+
+from dataclasses import asdict
+from logging import warning
+
+from ..config import LLMConfigurationError
+from ..errors import LLMImportError
+from . import LLMClient
+from .base import ChatMessage
+
+try:
+    from mistralai.client import MistralClient as _MistralClient
+    from mistralai.models.chat_completion import ChatMessage as MistralChatMessage
+except ImportError as err:
+    raise LLMImportError(
+        flavor="llm", msg="To use Mistral models, please install the `mistralai` package with `pip install mistralai`"
+    ) from err
+
+
+class MistralClient(LLMClient):
+    def __init__(self, model: str = "mistral-large-latest", client: _MistralClient = None):
+        self.model = model
+        self._client = client or _MistralClient()
+
+    def complete(
+        self,
+        messages: Sequence[ChatMessage],
+        temperature: float = 1.0,
+        max_tokens: Optional[int] = None,
+        caller_id: Optional[str] = None,
+        seed: Optional[int] = None,
+        format: str = None,
+    ) -> ChatMessage:
+        extra_params = dict()
+        if seed is not None:
+            extra_params["random_seed"] = seed
+
+        if format not in (None, "json", "json_object") and "large" not in self.model:
```

Comment:
format could be of `Literal` type

also pydantic's `validate_arguments` could do the validation job